internal/storage/crdb.go

```diff
@@ -35,31 +35,39 @@ func newCRDBEngine(config Config) (*crdbEngine, error) {
 		return nil, err
 	}
 
-	out := &crdbEngine{
+	out := &engine{
 		issuerService:      issSvc,
 		userInfoService:    userInfoSvc,
 		oauthClientManager: oauthClientManager,
 		db:                 db,
 	}
 
+	for _, opt := range options {
+		err = opt(out)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return out, nil
 }
 
-func (eng *crdbEngine) Shutdown() {
+func (eng *engine) Shutdown() error {
+	return eng.db.Close()
```

Comment:
I think we can kill this part of the interface now - the docs don't say it's necessary to do close this connection explicitly https://pkg.go.dev/database/sql#DB.Close

Reply:
That's probably fine; in that case we should have logic in place to better handle signals so we can wait for some amount of time before exiting.